pandas/core/groupby.py

```diff
@@ -3964,13 +4012,117 @@ def filter(self, func, dropna=True, *args, **kwargs):  # noqa
 
 class DataFrameGroupBy(NDFrameGroupBy):
     _apply_whitelist = _dataframe_apply_whitelist
```

Comment:
See tests/groupby/test_whitelist.py

Looks like it also shows up in `_dir_additions`:
```
def _dir_additions(self):
        return self.obj._dir_additions() | self._apply_whitelist
```

AFAICT `_apply_whitelist` is a historical artifact that may not be needed anymore, but is not actively causing any problems at the moment.